glean-core/csharp/Glean/Metrics/PingType.cs

```diff
@@ -0,0 +1,57 @@
+﻿// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+using Mozilla.Glean.FFI;
+using System;
+using static Mozilla.Glean.Glean;
+
+namespace Mozilla.Glean.Private
+{
+    /// <summary>
+    ///  An enum with no values for convenient use as the default set of reason codes.
+    /// </summary>
+    public enum NoReasonCodes : int
+    {
+    }
+
+    public class PingTypeBase
+    {
+        internal string name;
```

Comment:
Can you add a private constructor to make this class not instantiable on its own?

Reply:
We can make it as an `internal` constructor at least because it is inherited by other public classes.

Reply:
Oh, apparently we can further restrict this by making it `protected internal` (https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/protected-internal)